src/modules/editor/SoundEditor/Note.ts

```diff
@@ -0,0 +1,57 @@
+
+import * as Tone from "tone";
+import { SampleLibrary } from "@modules/editor/SoundEditor/Tonejs-Instruments"
+
+
+
+class Note {
+  public samp: Tone.Sampler;
+  constructor(
+    public note: string = "Nan",
+    private _duration: number = 1,
+    public instrument: string = "piano",
+  ) {
+    this.note = note;
+    this._duration = _duration;
+    this.instrument = instrument;
+    if (this.note == "Nan") {
+      this.samp = new Tone.Sampler();
+    } else {
+      this.samp = SampleLibrary.load({
+        instruments: this.instrument,
+        /* onload: () => {
+          console.log("Sampler is fully loaded!");
+        } */
+
+      }).toDestination();
+      //console.log(this.samp);
+    }
+
+  }
+
+  public get duration() {
+    return this._duration;
+  }
+
+  public toJson() {
```

Comment:
The function name is misleading. I think the best strategy here is to declare an interface in this class holding those items in one member, and then return this concrete structure when necessary.

This will avoid the GC from overrunning, especially with the amount of notes that can be present in a single sheet of music!